src/index.js

```diff
@@ -0,0 +1,129 @@
+import './main.css';
+
+// Vars
+const form = document.querySelector('#todoform');
+const todoInput = document.querySelector('#newtodo');
+const todosListEl = document.querySelector('#todos-list');
+
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let editTodoId = -1;
+let todoId = -100;
+
+// functions
+function renderTodo() {
+  // Clear render
+  todosListEl.innerHTML = '';
+  // render it
+  todos.forEach((todo, index) => {
+    todosListEl.innerHTML += `
+    <div class="todo" id="${index}">
+    <i class=""></i>
+    <i class="bi ${todo.checked ? 'bi-check-circle-fill' : 'bi-circle'} ""
+    data-action="check"
+    ></i>
+    <p class="paragraph ${todo.checked ? 'line' : ''}" data-action="check">${
+  todo.value
+}</p>
+    <i class="bi bi-pencil-square" data-action="edit"></i>
+    <i class="bi bi-trash" data-action="delete"></i>
+  </div>
+    `;
+  });
+}
+
+// save
+function saveTodos() {
+  const todoValue = todoInput.value;
+
+  // check if empty
+  const isEmpty = todoValue === '';
+
+  // Check for duplicate todos
+  const isDuplicate = todos.some(
+    (todo) => todo.value.toLowerCase() === todoValue.toLowerCase(),
+  );
+
+  if (isEmpty) {
+    alert('Todo is empty');
+  } else if (isDuplicate) {
+    alert('Todo already exist');
+  } else {
+    if (editTodoId >= 0) {
+      todos = todos.map((todo, index) => ({
+        ...todo,
+        value: index === editTodoId ? todoValue : todo.value,
+      }));
+      editTodoId = -1;
+    } else {
+      todos.push({
+        value: todoValue,
+        checked: false,
+      });
+    }
+
+    todoInput.value = '';
+  }
+}
```

Comment:
- [ ] You did a good job working on this milestone! 💯 However, according to the requirements:

![image](https://github.com/husush23/todo-list/assets/77069011/c8e22bc8-7cf0-4aae-89ec-710f3dd7f5c8)

Please consider moving the new functionality to a new Js file. 🤓 🔧